src/containers/Faucet/components/Form/index.tsx

```diff
@@ -0,0 +1,187 @@
+import { useState, useRef } from 'react'
+import { Box, Button, Stack, Typography } from '@mui/material'
+import ReCaptcha from 'react-google-recaptcha'
+import Card from 'components/Card'
+import axios from 'axios'
+import { StyledTextField } from 'components/Dialog/components/styles'
+import BigNumber from 'bignumber.js'
+import { useNotifications } from 'components/NotificationPopup/hooks'
+import { FaucetStatus } from 'store/faucetModal'
+import useModal from '../FaucetModal/hooks'
+import { styles } from './styles'
+import CosmosNetworkConfig from '../../../../ledgers/CosmosNetworkConfig'
+
+const Form = () => {
+  const captchaRef = useRef<any>(null)
+  const [address, setAddress] = useState<string>('')
+  const [amount, setAmount] = useState<string>('')
+  const [showTransferBtn, setShowTransferBtn] = useState<boolean>(false)
+  const { setWarning } = useNotifications()
+
+  const { handleModal } = useModal()
+
+  const handleAddress = (
+    ev: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    setAddress(ev.target.value)
+  }
+
+  const handleAmount = (
+    ev: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    setAmount(ev.target.value)
+  }
+
+  const checkCaptcha = async () => {
+    const token = captchaRef.current.getValue()
+
+    if (!amount) {
+      captchaRef.current.reset()
+    }
+    if (token) {
+      setShowTransferBtn(true)
+    } else {
+      setShowTransferBtn(false)
+    }
+  }
+
+  const handleReceiveTokens = async () => {
+    const token = captchaRef.current.getValue()
+
+    const transferAmount = new BigNumber(amount)
+      .multipliedBy(CosmosNetworkConfig.CURRENCY_1_CUDO)
+      .toString(10)
+
+    const re = /^cudos[0-9a-z]{39}$/
+    const correctCudosAddress = address.match(re)
+
+    if (!correctCudosAddress) {
+      setWarning('Wrong CUDOS Address Format')
+      return
+    }
+
+    const data = {
+      address,
+      coins: [`${transferAmount}${CosmosNetworkConfig.CURRENCY_DENOM}`],
+      captchaResponse: token
+    }
+
+    try {
+      handleModal({ open: true, status: FaucetStatus.LOADING })
+      const response = await axios
+        .post(import.meta.env.VITE_FAUCET_URL, data)
+        .then((res) => {
```

Comment:
this logic also looks a bit complex, can we make it simpler

Reply:
done